From fda1dc76a9c5209f5accb16dc96cb08fdddc5b90 Mon Sep 17 00:00:00 2001
From: Jackie Han <jackie.han@gmail.com>
Date: Thu, 29 Oct 2026 12:09:53 +0800
Subject: [PATCH] Show app versions under the row title

The version label in front of the update buttons has a different width on every row,
so the buttons start at different x positions. Show the version under the
"xxx Version" title instead, so the buttons always start in the same column.
---
 .../view/passwall2/app_update/app_version.htm    | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/luci-app-passwall2/luasrc/view/passwall2/app_update/app_version.htm b/luci-app-passwall2/luasrc/view/passwall2/app_update/app_version.htm
index aa3c0681..683d7293 100644
--- a/luci-app-passwall2/luasrc/view/passwall2/app_update/app_version.htm
+++ b/luci-app-passwall2/luasrc/view/passwall2/app_update/app_version.htm
@@ -312,11 +312,21 @@ local com = self.com
 //]]>
 </script>
 
+<style>
+	#app_version .app-version-sub {
+		display: block;
+		font-size: .85em;
+		font-weight: normal;
+		opacity: .7;
+	}
+</style>
+
 <div class="cbi-value">
-	<label class="cbi-value-title">Passwall2 <%:Version%></label>
+	<label class="cbi-value-title">Passwall2 <%:Version%>
+		<span class="app-version-sub"><%=api.get_version()%></span>
+	</label>
 	<div class="cbi-value-field">
 		<div class="cbi-value-description">
-			<span>【 <%=api.get_version()%> 】</span>
 			<input class="btn cbi-button cbi-button-apply" type="button" id="<%=api.appname%>-check_btn" data-update-app="<%=api.appname%>"
 				onclick="onBtnClick(this, appname);" value="<%:Check update%>" />
 			<input class="btn cbi-button cbi-button-apply" type="button" id="_<%=api.appname%>-force_btn" data-update-app="<%=api.appname%>"
@@ -332,10 +342,10 @@ local com = self.com
 <div class="cbi-value">
 	<label class="cbi-value-title"><%=v.name%>
 		<%:Version%>
+		<span class="app-version-sub"><span class="com_version" key="<%=k%>"><%=translate("Null")%></span></span>
 	</label>
 	<div class="cbi-value-field">
 		<div class="cbi-value-description">
-			<span>【 <span class="com_version" key="<%=k%>"><%=translate("Null")%></span> 】</span>
 			<input class="btn cbi-button cbi-button-apply" type="button" id="_<%=k%>-check_btn" data-update-app="<%=k%>"
 				onclick="onBtnClick(this,'<%=k%>');" value="<%:Check update%>" />
 			<input class="btn cbi-button cbi-button-apply" type="button" id="_<%=k%>-force_btn" data-update-app="<%=k%>"
-- 
2.43.0
